nvOC_miner_update.sh

```diff
@@ -25,354 +134,828 @@ function restart-if-needed {
+    fi
+    if [[ -L "${NVOC_MINERS}/TPccminer/recommended" && -d "${NVOC_MINERS}/TPccminer/recommended" ]]
+    then
+      rm ${NVOC_MINERS}/TPccminer/recommended
+    else
+      rm -rf ${NVOC_MINERS}/TPccminer/recommended
+    fi
+    ln -s "${NVOC_MINERS}/TPccminer/${TPccminer_9_ver}" "${NVOC_MINERS}/TPccminer/recommended"
+    ln -s "${NVOC_MINERS}/TPccminer/${TPccminer_9_ver}" "${NVOC_MINERS}/TPccminer/latest"
+  fi
   restart-if-needed
 else
-  echo "KTccminer-cryptonight is already up-to-date"
+  echo "tpruvot ccminer for CUDA-9.2 is already up-to-date"
+  echo "Use TPccminer_VERSION latest or recommended or ${TPccminer_9_ver} in 1bash"
 fi
 
+echo
 echo
 
-echo "Checking Equihash Bminer 9.1.0"
-if [ ! $(cat ${NVOC_MINERS}/bminer/latest/version | grep v9.1.0) ]
+echo "Checking Vertminer ${vertminer_8_ver}"
+if ! grep -q "${vertminer_8_ver}" ${NVOC_MINERS}/vertminer/${vertminer_8_ver}/version
 then
-  echo "Extracting Bminer"
-  mkdir -p ${NVOC_MINERS}/bminer/latest/
-  stop-if-needed "[b]miner"
-  cat ${NVOC_MINERS}/bminer/bminer-v9.1.0.tar.xz | tar -xJC ${NVOC_MINERS}/bminer/latest/ --strip 1
-  chmod a+x ${NVOC_MINERS}/bminer/latest/bminer
+  echo "Extracting vertminer"
+  mkdir -p ${NVOC_MINERS}/vertminer/${vertminer_8_ver}/
+  stop-if-needed "[v]ertminer"
+  tar -xvJf ${NVOC_MINERS}/vertminer/${vertminer_8_tarball} -C ${NVOC_MINERS}/vertminer/${vertminer_8_ver}/ --strip 1
+  chmod a+x ${NVOC_MINERS}/vertminer/${vertminer_8_ver}/vertminer
+  if [[ -L "${NVOC_MINERS}/vertminer/recommended" && -d "${NVOC_MINERS}/vertminer/recommended" ]]
+  then
+    rm ${NVOC_MINERS}/vertminer/recommended
+  else
+    rm -rf ${NVOC_MINERS}/vertminer/recommended
+  fi
+  if [[ -L "${NVOC_MINERS}/vertminer/latest" && -d "${NVOC_MINERS}/vertminer/latest" ]]
+  then
+    rm ${NVOC_MINERS}/vertminer/latest
+  else
+    rm -rf ${NVOC_MINERS}/vertminer/latest
+  fi
+  ln -s "${NVOC_MINERS}/vertminer/${vertminer_8_ver}" "${NVOC_MINERS}/vertminer/recommended"
+  ln -s "${NVOC_MINERS}/vertminer/${vertminer_8_ver}" "${NVOC_MINERS}/vertminer/latest"
   restart-if-needed
 else
-  echo "Bminer is already up-to-date"
+  echo "Vertminer is already up-to-date"
 fi
 
+echo
 echo
 
-echo "Checking ANXccminer (git@cd6fab68823e247bb84dd1fa0448d5f75ec4917d)"
-if [ ! $(cat ${NVOC_MINERS}/ANXccminer/version | grep cd6fab68823e247bb84dd1fa0448d5f75ec4917d) ]
+echo "Checking xmr_stak 2.4.4"
+if ! grep -q "2.4.4" ${NVOC_MINERS}/xmr_stak/2.4.4/version
 then
-  echo "Extracting ANXccminer"
-  mkdir -p ${NVOC_MINERS}/ANXccminer/
-  stop-if-needed "[A]NXccminer"
-  cat ${NVOC_MINERS}/ANXccminer/ANXccminer.tar.xz | tar -xJC ${NVOC_MINERS}/ANXccminer/ --strip 1
-  chmod a+x ${NVOC_MINERS}/ANXccminer/ccminer
+  echo "Extracting xmr-stak"
+  mkdir -p ${NVOC_MINERS}/xmr_stak/2.4.4/
+  stop-if-needed "[x]mr-stak"
+  tar -xvJf ${NVOC_MINERS}/xmr_stak/xmr-stak-2.4.4.tar.xz -C ${NVOC_MINERS}/xmr_stak/2.4.4/ --strip 1
+  chmod a+x ${NVOC_MINERS}/xmr_stak/2.4.4/xmr_stak_miner
```

Comment:
I need the executable to have the same as it comes out of the compilation process, please rename this executable here and inside the tarball to "xmr-stak", I did it already on the main nvOC repo. Keep the "_" in the folder name as it is needed  for proper variables expansion. Use xmr_stak instead of xmr-stak also for the tarball filename.

Reply:
All miners executables should have miner in name so watchdog, 3main, .... Can catch  running miner.

Reply:
WTM also check miner executable running time

Reply:
Should catch fine also on path names, please check

Reply:
Should catch fine because they still match the screen session name which is "miner".